components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/APIUsageStatisticsClient.java

```diff
@@ -2886,6 +2887,399 @@ private OMElement queryFirstAccess(String columnFamily)
         }
     }
 
+    /** Given API name and Application, returns throttling request counts over time for a given time span
+     * 
+     * @param apiName Name of the API
+     * @param provider Provider name
+     * @param appName Application name
+     * @param fromDate Start date of the time span
+     * @param toDate End date of time span
+     * @param groupBy Group by parameter. Supported parameters are :day,hour
+     * @return Throttling counts over time
+     * @throws APIMgtUsageQueryServiceClientException
+     */
+    public List<APIThrottlingOverTimeDTO> getThrottleDataOfAPIAndApplication(String apiName, String provider,
+            String appName, String fromDate, String toDate, String groupBy)
+            throws APIMgtUsageQueryServiceClientException {
+
+        if (dataSource == null) {
+            throw new APIMgtUsageQueryServiceClientException("BAM data source hasn't been initialized. Ensure " +
+                    "that the data source is properly configured in the APIUsageTracker configuration.");
+        }
```

Comment:
Please note the SQL injection issue is already fixed in 9cc1b0ae6d4b3c1d8079416d1ae987e6b11bfa7f